modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/CET.java

```diff
@@ -29,6 +30,8 @@ public interface CET {
 
 	public String getBaseURL();
 
+	public ClientExtensionEntry getClientExtensionEntry();
```

Comment:
Isn't this a bit leaky? `CET` are not always backed up by a `ClientExtensionEntry`. They may come from an OSGi configuration, for example...

I'm sure there must be another way to do it. Or at least we can declare a new interface with this method and test the object instances with `instanceof TheNewInterface` to make it a bit safer...

Reply:
Now looking at the LPS, it seems that the real problem you are having is to register portlets coming from client extensions with unique names. If that is the case, then I guess we should fix it in the CET Deployer, and add there something to make portlet ids unique.

Otherwise, we will probably have the same problem for LXC because, as I said, `CET`s coming from LXC don't have a `ClientExtensionEntry`.

Reply:
Feel free to send the PR to `liferay-frontend` user and we can see if we can find a better solution there. :+1:

Reply:
Seen https://github.com/liferay-frontend/liferay-portal/pull/2650, will continue there.